packages/access-api/test/provider-add.test.js

```diff
@@ -97,6 +99,100 @@ for (const accessApiVariant of /** @type {const} */ ([
       })
     })
   })
+
+  if (
+    ['provider/add', 'access/delegate'].every((f) =>
+      accessApiVariant.features.has(f)
+    )
+  ) {
+    it('provider/add allows for access/delegate', async () => {
+      const space = await principal.ed25519.generate()
+      const issuer = await accessApiVariant.issuer
+      const service = await accessApiVariant.audience
+      const accountDid = /** @type {const} */ ('did:mailto:example.com:foo')
+      const serviceSessionAttest = await ucanto.delegate({
+        issuer: service,
+        audience: issuer,
+        capabilities: [
+          {
+            can: 'provider/add',
+            with: accountDid,
+          },
+          {
+            can: 'access/delegate',
+            with: space.did(),
+          },
+        ],
+      })
+      const sessionProofs = [
+        await ucanto.delegate({
+          issuer: service,
+          audience: issuer,
+          capabilities: [
+            {
+              can: 'ucan/attest',
+              with: accountDid,
+              nb: {
+                // note: whole delegation is also included in 'proofs'
+                proof: serviceSessionAttest.cid,
+              },
+            },
+          ],
+        }),
+        serviceSessionAttest,
+      ]
+      const addStorageProvider = await ucanto
+        .invoke({
+          issuer,
+          audience: service,
+          capability: {
+            can: 'provider/add',
+            with: accountDid,
+            nb: {
+              provider: 'did:web:web3.storage:providers:w3up-alpha',
+              consumer: space.did(),
+            },
+          },
+          proofs: [
+            // space says issuer can provider/add with this account
```

Comment:
Hmm... But space should have no authority over the account here.

Reply:
🤦 good call
https://github.com/web3-storage/w3protocol/pull/483/commits/90afea4e7eb68353d0938e5adda76fe0cd8a12c0